cpp/src/arrow/compute/exec/hash_join_node.cc

```diff
@@ -294,8 +294,7 @@ std::shared_ptr<Schema> HashJoinSchema::MakeOutputSchema(
         proj_maps[side].data_type(HashJoinProjection::INPUT, input_field_id);
 
     std::string output_field_name =
-        (is_left ? left_field_name_prefix : right_field_name_prefix) + input_field_name;
-
+        input_field_name + (is_left ? left_field_name_prefix : right_field_name_prefix);
```

Comment:
Should these variables be renamed to `*_field_name_suffix`?

Reply:
Yes, I think there are a few used in a few places that I have missed.